controllers/operator/manager.go

```
@@ -466,36 +466,73 @@ func (m *ODLMOperator) GetSubscription(ctx context.Context, name, operatorNs, se
 func (m *ODLMOperator) GetDeployment(ctx context.Context, name, operatorNs, servicesNs, packageName string) (*appsv1.Deployment, error) {
 	klog.V(3).Infof("Fetch Deployment %s (package name: %s) in operatorNamespace %s and servicesNamespace %s", name, packageName, operatorNs, servicesNs)
 
-	tenantScope := make(map[string]struct{})
-	for _, ns := range []string{operatorNs, servicesNs} {
-		tenantScope[ns] = struct{}{}
-	}
-
-	var depCandidates []appsv1.Deployment
-	for ns := range tenantScope {
-		depList := &appsv1.DeploymentList{}
-		if err := m.Client.List(ctx, depList, &client.ListOptions{
-			Namespace: ns,
-		}); err != nil {
-			return nil, err
-		}
-
-		for _, dep := range depList.Items {
-			if dep.Name == name || dep.Annotations["packageName"] == packageName {
-				depCandidates = append(depCandidates, dep)
-			}
+	deploy := &appsv1.Deployment{}
```

Doesn't OLM-based ODLM use the `GetDeployment` function? If so, all of the changes in this function need to be separated out into a function dedicated for non-OLM

---

I'm assuming those tests that are failing are probably from the changes in this function

---

Right, this `GetDeployment` function is newly created and only used for the non-OLM controller. But since the deployment names are quite different, I will remove it and use `GetDeploymentListFromPackage` instead to find the list of deployments and locate the intended one.

`GetDeploymentListFromPackage` has been used in several places, only in the non-OLM controller. I think it could also used as a re-req check for ODLM to verify if the required deployment is present. If not, it should return an error